camayoc/tests/qcs/api/v1/scans/test_satellite_scans.py

```diff
@@ -0,0 +1,167 @@
+# coding=utf-8
+"""Tests for ``Scan`` API endpoint for quipucords server.
+
+:caseautomation: automated
+:casecomponent: api
+:caseimportance: high
+:caselevel: integration
+:requirement: Sonar
+:testtype: functional
+:upstream: yes
+"""
+
+import pytest
+
+from camayoc import config
+from camayoc.exceptions import ConfigFileNotFoundError
+from camayoc.qcs_models import (
+    Credential,
+    Source,
+    Scan
+)
+from camayoc.tests.qcs.api.v1.scans.utils import wait_until_state
+
+
+TIMEOUT = 540
+
+
+def sat_source():
+    """Gather Satellite source from config file.
+
+    If no config file is found, or no sources defined,
+    then an empty list will be returned.
+
+    If a test is parametrized on the sources in the config file, it will skip
+    if no source is found.
+    """
+    try:
+        src = [config.get_config()['satellite']]
+    except (ConfigFileNotFoundError, KeyError):
+        src = []
+    return src
+
+
+def prep_broken_sat_scan(cleanup):
+    """Return a scan that can be created but will fail to complete.
+
+    Create and return a source with a non-existent host and dummy credential.
+    It is returned ready to be POSTed to the server via the create() instance
+    method.
+    """
+    bad_cred = Credential(
+        username='broken',
+        password='broken',
+        cred_type='satellite'
+    )
+    bad_cred.create()
+    cleanup.append(bad_cred)
+    bad_src = Source(
+        source_type='satellite',
+        hosts=['1.0.0.0'],
+        credential_ids=[bad_cred._id],
+    )
+    bad_src.create()
+    cleanup.append(bad_src)
+    bad_scn = Scan(
+        source_ids=[bad_src._id],
+    )
+    return bad_scn
+
+
+def prep_sat_scan(source, cleanup, client):
+    """Given a source from config file, prep the scan.
+
+    Takes care of creating the Credential and Source objects on the server and
+    staging them for cleanup after the tests complete.
+    """
+    cred_ids = []
+    cred = Credential(
+        cred_type='sat',
+        client=client,
+        password=source['password'],
+        username=source['username'],
+    )
+    cred.create()
+    cleanup.append(cred)
+    cred_ids.append(cred._id)
+
+    satsrc = Source(
+        source_type='sat',
+        client=client,
+        hosts=[source['hostname']],
+        credential_ids=cred_ids,
+    )
+    satsrc.create()
+    cleanup.append(satsrc)
+    scan = Scan(source_ids=[satsrc._id])
+    return scan
+
+
+@pytest.mark.skip
```

Comment:
Are you skipping this because any open bug? If yes I think it is better to xfail this and as the reason pass the bug URL. Check CLI credentials code, there is an example there.

Reply:
OK I can look to see if there is a issue I can reference that will make these start running when the satellite creds/sources/scans get implemented. I was skipping because they are not yet implemented

Reply:
I see, if it is not implemented I think you can skip them but also make them not automated